src/file-manager/fm-desktop-icon-view.c

```diff
@@ -169,6 +183,8 @@ net_workarea_changed (FMDesktopIconView *icon_view,
      * elements in the workareas array describing
      * x,y,width,height) */
     display = gtk_widget_get_display (GTK_WIDGET (icon_view));
+    if  (!GDK_IS_X11_DISPLAY (display))
+       return;
```

Comment:
This whole function does nothing but on X11, so you should remove this and rather make the function itself (`net_workarea_changed()`) only called on X11

Reply:
Turns out I can remove line 186 and 187 w no other changes

This is called in threee places total,  all already behind x11-only conditionals. One is in
`desktop_icon_view_property_filter()` which in turn is called only from inside an x11-only block in 
`realized_callback () `and another x11-only block in `unrealized_callback ()`

It is also called directly from the same x11-only block in 
`realized_callback () ` in which `desktop_icon_view_property_filter()` is called
which means when running x11 it gets called twice, only five non-blank lines apart
and in wayland is not reached at all. The x11 codepath is same as before, just put
into the x11 only conditional.

Asserts could be used to catch future programming errors calling the x11-only code
from wayland, the comments would tell what functions to put them in. That's still more code to run however